41-vaccine-temp-monitor/firmware/arduino/src/main.cpp

```diff
@@ -0,0 +1,528 @@
+        if (!notecard.sendRequest(cmd))
+        {
+            JDelete(cmd);
+        }
+    }
+
+    va_end(args);
+}
+
+int acquireGPSLocation(size_t timeout_s_ = 95)
+{
+    int result;
+    const char *gps_mode;
+    size_t gps_time_s;
+
+    // Get Current Configuration
+    if (J *rsp = notecard.requestAndResponse(notecard.newRequest("card.location.mode")))
+    {
+        gps_mode = JGetString(rsp, "mode");
+        NoteDeleteResponse(rsp);
+    }
+    else
+    {
+        result = -1; // Unable to communicate with Notecard
+        return result;
+    }
+
+    // Halt periodic tracking, before active sync is performed
+    if (J *cmd = notecard.newCommand("card.location.mode"))
+    {
+        JAddStringToObject(cmd, "mode", "off");
+        notecard.sendRequest(cmd);
+    }
+
+    // Gather timestamp of previous location information
+    if (J *rsp = notecard.requestAndResponse(notecard.newRequest("card.location")))
+    {
+        gps_time_s = JGetInt(rsp, "time");
+        logNoteF("INFO: Previous GPS time (seconds): %u", gps_time_s);
+        NoteDeleteResponse(rsp);
+    }
+    else
+    {
+        result = -1; // Unable to communicate with Notecard
+        return result;
+    }
+
+    // Activate GPS
+    if (J *cmd = notecard.newCommand("card.location.mode"))
+    {
+        JAddStringToObject(cmd, "mode", "continuous");
+        notecard.sendRequest(cmd);
+    }
+
+    // Block while resolving GPS location
+    for (const size_t start_ms = ::millis();;)
+    {
+        if (timeout_s_)
+        {
+            if (::millis() >= (start_ms + (timeout_s_ * 1000)))
+            {
+                logNoteF("WARNING: GPS user timeout expired!");
+                result = -3; // User timeout has expired
+                break;
+            }
+        }
+
+        // Check if GPS has acquired location information
+        if (J *rsp = notecard.requestAndResponse(notecard.newRequest("card.location")))
+        {
+            const size_t current_gps_time_s = JGetInt(rsp, "time");
+            if (current_gps_time_s != gps_time_s)
+            {
+                logNoteF("INFO: Current GPS time (seconds): %u", current_gps_time_s);
+                result = 0; // GPS has fixed on position
+                break;
+            }
+            if (JGetObjectItem(rsp, "stop"))
+            {
+                logNoteF("WARNING: GPS internal timeout expired!");
+                result = -2; // Notecard has signaled polling should stop,
+                             // most likely caused by internal GPS timeout
+                break;
+            }
+            NoteDeleteResponse(rsp);
+        }
+        else
+        {
+            result = -1; // Unable to communicate with Notecard
+            break;
+        }
+        ::delay(2500);
+    }
+
+    // Restore Previous Configuration
+    if (J *cmd = notecard.newCommand("card.location.mode"))
+    {
+        JAddStringToObject(cmd, "mode", gps_mode);
+        notecard.sendRequest(cmd);

```

Comment:
Check return.

Reply:
We decided offline that checking a lot of these isn't super necessary. Any errors will come back from the Notecard's debug logging (after changing from commands to requests).